frontend/components/forms/admin/AppConfigForm/AppConfigFormFunctional.tsx

```diff
@@ -0,0 +1,1121 @@
+
+import Button from "components/buttons/Button";
+import Checkbox from "components/forms/fields/Checkbox";
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import EnrollSecretTable from "components/EnrollSecretTable";
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+// @ts-ignore
+import OrgLogoIcon from "components/icons/OrgLogoIcon";
+// @ts-ignore
+import validate from "components/forms/admin/AppConfigForm/validate";
+import IconToolTip from "components/IconToolTip";
+import InfoBanner from "components/InfoBanner/InfoBanner";
+// @ts-ignore
+import YamlAce from "components/YamlAce";
+import Modal from "components/Modal";
+import OpenNewTabIcon from "../../../../../assets/images/open-new-tab-12x12@2x.png";
+import { string } from "prop-types";
+import SelectTargetsDropdownStories from "components/forms/fields/SelectTargetsDropdown/SelectTargetsDropdown.stories";
+
+const authMethodOptions = [
+  { label: "Plain", value: "authmethod_plain" },
+  { label: "Cram MD5", value: "authmethod_cram_md5" },
+  { label: "Login", value: "authmethod_login" },
+];
+const authTypeOptions = [
+  { label: "Username and Password", value: "authtype_username_password" },
+  { label: "None", value: "authtype_none" },
+];
+const percentageOfHosts = [
+  { label: "1%", value: 1 },
+  { label: "5%", value: 5 },
+  { label: "10%", value: 10 },
+  { label: "25%", value: 25 },
+];
+const numberOfDays = [
+  { label: "1 day", value: 1 },
+  { label: "3 days", value: 3 },
+  { label: "7 days", value: 7 },
+  { label: "14 days", value: 14 },
+];
+
+// TODO: consider breaking this up into separate components/files
+
+const baseClass = "app-config-form";
+
+interface IAppConfigFormProps {
+  formData: IConfigNested;
+  enrollSecret: IEnrollSecret[] | undefined;
+  handleSubmit: any;
+}
+
+interface IFormField {
+  name: string;
+  value: string | boolean | number;
+}
+
+interface IAppConfigFormErrors {
+  metadata_url?: string | null;
+  entity_id?: string | null;
+  idp_name?: string | null;
+  server_url?: string | null;
+  org_name?: string | null;
+  sender_address?: string | null;
+  server?: string | null;
+  user_name?: string | null;
+  password?: string | null;
+  destination_url?: string | null;
+  host_percentage?: string | null;
+  days_count?: string | null;
+  host_expiry_window?: string | null;
+  agent_options?: string | null;
+}
+
+const AppConfigFormFunctional = ({
+  formData,
+  enrollSecret,
+  handleSubmit,
+}: IAppConfigFormProps): JSX.Element => {
+  // █▀ ▀█▀ ▄▀█ ▀█▀ █▀▀
+  // ▄█ ░█░ █▀█ ░█░ ██▄
+  const [
+    showHostStatusWebhookPreviewModal,
+    setShowHostStatusWebhookPreviewModal,
+  ] = useState<boolean>(false);
+  const [
+    showUsageStatsPreviewModal,
+    setShowUsageStatsPreviewModal,
+  ] = useState<boolean>(false);
+
+  // █▀▀ █▀█ █▀█ █▀▄▀█   █▀ ▀█▀ ▄▀█ ▀█▀ █▀▀
+  // █▀░ █▄█ █▀▄ █░▀░█   ▄█ ░█░ █▀█ ░█░ ██▄
+  const [iterateFormData, setIterateFormData] = useState<any>({
```

Comment:
The name `iterateFormData` is confusing to me. I'm not sure why iterate is including. Could it be just `formData`?

Reply:
This object should also be an interface so you don't have to use `any`.